pkg/prometheus/promcfg.go

```diff
@@ -444,6 +444,38 @@ func stringMapToMapSlice[V any](m map[string]V) yaml.MapSlice {
 	return res
 }
 
+func mergeSafeAuthorizationWithScrapeClass(authz *monitoringv1.SafeAuthorization, scrapeClass monitoringv1.ScrapeClass) *monitoringv1.Authorization {
+	if authz == nil || reflect.ValueOf(*authz).IsZero() {
+		return mergeAuthorizationWithScrapeClass(nil, scrapeClass)
+	}
+
+	return mergeAuthorizationWithScrapeClass(&monitoringv1.Authorization{SafeAuthorization: *authz}, scrapeClass)
+}
+
+func mergeAuthorizationWithScrapeClass(authz *monitoringv1.Authorization, scrapeClass monitoringv1.ScrapeClass) *monitoringv1.Authorization {
+	if authz == nil {
+		return scrapeClass.Authorization
+	}
+
+	if scrapeClass.Authorization == nil {
+		return authz
+	}
+
+	if authz.SafeAuthorization.Credentials == nil {
+		authz.SafeAuthorization.Credentials = scrapeClass.Authorization.SafeAuthorization.Credentials
+	}
+
+	if authz.Credentials == nil {
```

Comment:
can we also check that `authz.CredentialsFile == ""` since Credentials and CredentialsFile are (should) be mutually exclusive?

Reply:
shall we take care of that here and catch it more "globally" since it seems to be already allowed here

```go
func (cg *ConfigGenerator) addAuthorizationToYaml(
	cfg yaml.MapSlice,
	store assets.StoreGetter,
	auth *monitoringv1.Authorization,
) yaml.MapSlice {
	if auth == nil {
		return cfg
	}

	// reuse addSafeAuthorizationToYaml and unpack the part we're interested
	// in, namely the value under the "authorization" key
	authCfg := cg.addSafeAuthorizationToYaml(yaml.MapSlice{}, store, &auth.SafeAuthorization)[0].Value.(yaml.MapSlice)

	if auth.CredentialsFile != "" {
		authCfg = append(authCfg, yaml.MapItem{Key: "credentials_file", Value: auth.CredentialsFile})
	}

	return cg.WithMinimumVersion("2.26.0").AppendMapItem(cfg, "authorization", authCfg)
}
```

Reply:
It would definitely good to add (as a follow-up PR). I noticed the same when reviewing this change :)